rules/order/src/Rector/ClassMethod/OrderConstructorDependenciesByTypeAlphabeticallyRector.php

```diff
@@ -83,24 +83,26 @@ private function getSortedParams(ClassMethod $classMethod): array
     {
         $params = $classMethod->getParams();
         usort($params, function (Param $firstParam, Param $secondParam) {
-            /** @var Name $typeA */
-            $typeA = $firstParam->type;
-            /** @var Name $typeB */
-            $typeB = $secondParam->type;
-            return strcmp($this->getShortName($typeA), $this->getShortName($typeB));
+            /** @var Name $firstParamType */
+            $firstParamType = $this->getParamType($firstParam);
+            /** @var Name $secondParamType */
+            $secondParamType = $this->getParamType($secondParam);
+
+            return strcmp($this->getShortName($firstParamType), $this->getShortName($secondParamType));
```

Comment:
@TomasVotruba Just curious... is it okay that I'm making phpstan fool here? :D
`getParamType` returns `Identifier|Name|UnionType|null` but in this method I'm changing the type in annotation to `Name`

Reply:
I would not say fool, but rather share with him what you know :)
If that's 100 % the case there is a Name, then why not :+1: 

In time there might be an exception, but that's what mostly running rule on Rector code base revels.

Reply:
What about adding test case completely without type?

```php
public function __construct($whatever, stdClass $someObject)
{
}
```

Reply:
It is already there ;)

Reply:
https://github.com/rectorphp/rector/blob/ec6a9d8da05caecf2e3da1752c67ea02dc89d9ca/rules/order/tests/Rector/ClassMethod/OrderConstructorDependenciesByTypeAlphabeticallyRector/Fixture/skip_param_with_no_type.php.inc

Reply:
Btw, how does `strcmp` compare to `<=>`? I never used `strcmp`, as the name is not clear what it does for me, so I'm curious

Reply:
Changed to `<=>`